### ISSUE TITLE
Transform and broadcast assignment

### ISSUE THREAD

**Issue by jkoliner** [JuliaData/DataFramesMeta.jl#137]
It looks like the new method of broadcast assignment isn't mixing well with the transform macro. I don't think this is a bug so much as a decision that has yet to be made.
```
somedf = DataFrame(a=[1,2,3], b=[2,3,4])
## Doesn't work (used to work)
@transform(somedf, c = 0)
## Doesn't work
@transform(somedf, c.= 0)
​
## Doesn't work
somedf[!,:c] = 0
## Works
somedf[!,:c] .= 0

**Comment by konkam** [comment]
Hello, 

I am running into the same problem, `@transform(somedf, c = 0)` used to work and I found it fairly convenient. `@transform(somedf, c.= 0)` would certainly be an acceptable replacement, is there any alternative option than the verbose:

```
@transform(somedf, c = repeat([0], inner = length(:a))
```

?

**Comment by Byrth** [comment]
I also have this problem.

Does everyone have this problem?

**Comment by bkamins** [comment]
I think DataFramesMeta.jl is not actively maintained by anyone now. Feel free to make a PR fixing the issue and I can review it then.

**Comment by Byrth** [comment]
OK, opened a PR to this effect. It took me a bit to wrap my head around what was happening under the hood, but I think this is valid.
https://github.com/JuliaData/DataFramesMeta.jl/pull/142

**Comment by pdeffebach** [comment]
This should be fixed by #163

```
transform(df, t = 1)
```

will now work.

**Comment by bkamins** [comment]
```
@transform(somedf, c = 0)
```
works
